Added DisplaySettings interface for footer on/off via settings.json

oneness opened the pull request:
## TLDR

This pull request introduces a new configuration option, `display.footer`, which allows users to hide the footer/status bar for a more minimal UI. The default behavior remains unchanged, with the footer visible.

Key files to review:
*   `packages/core/src/config/config.ts`: For the new configuration schema.
*   `packages/cli/src/ui/components/Footer.tsx`: For the conditional rendering logic.
*   `packages/cli/src/ui/App.test.tsx`: For the corresponding test updates.

## Dive Deeper

To provide users with more control over the UI's verbosity, I've added a feature to conditionally render the footer.

The implementation involved three main steps:
1.  **Configuration Update**: I extended the configuration schema in `packages/core/src/config/config.ts` by adding a `display` object that can contain a `footer: boolean` property.
2.  **UI Modification**: The `Footer.tsx` component now receives the `display` configuration and will render `null` if `display.footer` is explicitly set to `false`.
3.  **Test Updates**: The unit tests for `App.tsx` were failing because the mock `Config` object was missing the new `getDisplay()` method. I updated `App.test.tsx` to include a mock for this method, ensuring all tests pass.

During development, I encountered and resolved a series of `npm run preflight` failures that were caused by a corrupt local npm cache, which required fixing file ownership before the checks could pass.

## Reviewer Test Plan

1.  Pull this branch and run `npm run preflight` to ensure all checks pass.
2.  Start the Gemini CLI normally (`npm start`). The footer should be visible by default.
3.  Open or create your user settings file at `~/.gemini/settings.json`.
4.  Add the following configuration to the file and save it:
    ```json
    {
      "display": {
        "footer": false
      }
    }
    ```
5.  Restart the Gemini CLI. The footer should now be completely hidden.
6.  To verify it reappears, either remove the `display` object from your settings or set `footer` to `true` and restart the CLI.

## Testing Matrix

|          | macOS | Linux | Windows |
| -------- | :---: | :---: | :---: |
| npm run  |   ✅   |  ❓   |   ❓   |
| npx      |   ❓   |  ❓   |   ❓   |
| Docker   |   ❓   |  ❓   |   ❓   |
| Podman   |   ❓   |   -   |   -   |
| Seatbelt |   ❓   |   -   |   -   |

## Linked issues
https://github.com/google-gemini/gemini-cli/issues/4509

oneness commented:
Hi, @jacob314
Hope this PR is ok with what you are thinking.

jacob314 commented on packages/core/src/config/config.ts:
```diff
@@ -56,6 +56,10 @@ export interface AccessibilitySettings {
   disableLoadingPhrases?: boolean;
 }
 
+export interface DisplaySettings {
```
rather than placing this under config.ts in the core package please place this in settings.ts in the cli package.

jacob314 commented on packages/cli/src/ui/components/Footer.tsx:
```diff
@@ -39,10 +41,15 @@ export const Footer: React.FC<FooterProps> = ({
   const limit = tokenLimit(model);
   const percentage = promptTokenCount / limit;
 
+  if (display.footer === false) {
```
one concern is that users of this setting will not be able to tell when errors have occurred. I would be more comfortable with the option if we still showed the number of errors that had occurred even when displayFooter is false.

jacob314 reviewed (commented):
Thanks for the contribution. I do think some users will appreciate the option of hiding the footer but we do need to make sure users who do so can still understand when gemini cli is experiencing serious errors.

jacob314 commented:
Note that the pull request description is a bit misleading. It is stating that this enables setting the option via settings.json while settings.ts must be used for settings configurable by settings.json. Did Gemini CLI gemerate that? If so, I'll be sure to update the GEMINI.md for the project to try to clarify how settings should be added.

jacob314 commented:
I'm so sorry I missed the notification that it was my turn to review this. I've now fixed my notification settings. If you still want to proceed just fix the merge conflicts and I'll approve.

oneness commented:
> I'm so sorry I missed the notification that it was my turn to review this. I've now fixed my notification settings. If you still want to proceed just fix the merge conflicts and I'll approve.

@jacob314 No worries. Things happen :).  I fixed the merge conflicts as you suggested and feel free to merge it in at your convenience.

jacob314 commented on packages/cli/src/config/settings.ts:
```diff
@@ -61,6 +69,19 @@ export interface AccessibilitySettings {
   screenReader?: boolean;
 }
 
+export interface DisplaySettings {
```
this merge didn't go correctly. Please update settingSchema.ts instead of add your additional option. Please also align with additional naming for settings. there are similar existing settings for hiding other portions of the ui.

jacob314 commented on GEMINI.md:
```diff
@@ -191,3 +191,13 @@ Only write high-value comments if at all. Avoid talking to the user through comm
 
 Use hyphens instead of underscores in flag names (e.g. `my-flag` instead of `my_flag`).
+
+## Configuration and Settings
```
please revert. this is no longer consistent with how settings.ts works. Sorry the code base moved fast so things get out of date quickly. :(

jacob314 reviewed (commented):
I'll approve once this comments are addressed. thanks again for the pr. expect many users will appreciate being able to hide the footer.
